src/services/LLMService.ts

```diff
@@ -93,57 +135,430 @@ export class LLMService {
+            apiUrl += '/v1/chat/completions';
+        }
+
+
+        const payload = {
+            model: 'llama3',
+            messages: [
+                {
+                    role: 'system',
+                    content: systemPrompt,
+                },
+                {
+                    role: 'user',
+                    content: query,
+                },
+            ],
+            temperature: 0.7,
+            max_tokens: 2000,
+        };
+
+        const request: IHttpRequest = {
+            headers: {
+                [HttpHeaders.CONTENT_TYPE]: ContentTypes.APPLICATION_JSON,
+            },
+            data: payload,
+        };
 
         try {
-            const response = await this.http.post(this.llmEndpoint, request);
+
+            
+            const response = await this.http.post(apiUrl, request);
 
             if (!response?.data) {
-                throw new Error('Failed to get response from LLM API');
+                throw new Error(t(Translations.LLM_NO_RESPONSE, this.language));
             }
 
             const llmResponse = response.data as ILLMResponse;
 
             if (!llmResponse.choices?.length) {
-                throw new Error('No choices in LLM response');
+                throw new Error(t(Translations.LLM_NO_CHOICES, this.language));
             }
 
             const choice = llmResponse.choices[0];
-            const content = choice.message.content || '';
-            return content.replace(/^Summary:|\bSummary\b:/i, "").trim() || "Failed to generate summary.";
+            let toolCalls = choice.message.tool_calls || [];
+            let error: string | undefined;
+
+            // Parse content for tool calls if no direct tool_calls
+            if (toolCalls.length === 0 && choice.message.content) {
+                const parsed = this.parseContentForTools(choice.message.content);
+                toolCalls = parsed.tools;
+                error = parsed.error;
+            }
+
+            return { toolCalls, rawResponse: llmResponse, error };
+        } catch (error) {
+            // Enhanced error logging for self-hosted LLM debugging
+            this.app.getLogger().error('Self-hosted LLM Error Details:', {
+                originalUrl: this.llmSettings.selfHostedUrl,
+                processedUrl: apiUrl,
+                error: error.message,
+                errorType: error.constructor.name,
+                statusCode: error.statusCode,
+                response: error.response?.data
+            });
             
+            const errorMessage = handleLLMErrorAndGetMessage(this.app, 'Self-hosted Provider', error, this.language);
+            throw new Error(errorMessage);
+        }
+    }
+
+    public async generateSummary(messages: string, channelName: string): Promise<string> {
+        const prompt = LlmPrompts.SUMMARIZE_PROMPT
+            .replace('__channelName__', channelName)
```

Comment:
make constants for all  strings that you use again and again `__channelName__` ,`__messages__`